adr/content-management/2023-01-06-upload-thumbnail-api-concept.md

```diff
@@ -0,0 +1,72 @@
+# 2023-01-06 - Upload Thumbnails API
+
+## Context
+We’re building an App that should optimize Thumbnails via the App. Currently, it’s not possible to upload Thumbnails via the API. All Thumbnails are processed by Shopware itself. There’s no way to upload your optimized Thumbnails via the API.
+
+However: Since it’s an App, we need to be able to do so via the API.
+
+## Decision
+### Migration
+Add a new column `isUploaded` into the media_thumbnail table. This flag lets us know if the thumbnail is uploaded or generated.
+
+### Core
+
+Add a new `updloadThumbnail` API in Shopware\Core\Content\Media\Api\MediaUploadController
+```php
+/**
+ * @Route("/api/_action/media/{mediaId}/upload-thumbnail", name="api.action.media.upload-thumbnail", methods={"POST"})
+ */
+public function uploadThumbnail(string $mediaId, Request $request, Context $context, ResponseFactoryInterface $responseFactory): Response
```

Comment:
How do you want to handle multiple thumbnails. Each thumbnail file is an API calls against this?

Reply:
I intend to do multiple uploads in one request to reduce the requests to Shopware. Which way do you think is better?

Reply:
It depends. A complete request should not be bigger than 10MB. So it't should be fine

Reply:
Okay, got it.